sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "opentelemetry/core/timestamp.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/sdk/trace/recordable.h"
+#include "opentelemetry/trace/canonical_code.h"
+#include "opentelemetry/trace/span_id.h"
+#include "opentelemetry/trace/trace_id.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+/**
+ * SpanData is an immutable representation of all data collected by a span.
+ */
+class SpanData final : public Recordable
+{
+public:
+  // The trace id for this span.
+  virtual opentelemetry::trace::TraceId GetTraceId() const noexcept = 0;
```

Comment:
Why are these functions virtual?

Reply:
Sorry, that was a relic from a previous thought process. I removed the `virtual`s.